netadapter/netadapter.go

```diff
@@ -0,0 +1,56 @@
+package netadapter
+
+import (
+	"github.com/kaspanet/kaspad/netadapter/server"
+	"github.com/kaspanet/kaspad/netadapter/server/grpc"
+)
+
+// NetAdapter is an adapter to the net
+type NetAdapter struct {
+	server            server.Server
+	routerInitializer func(peer *Peer) *Router
+}
+
+// NewNetAdapter creates and starts a new NetAdapter on the
+// given listeningPort
+func NewNetAdapter(listeningPort string) (*NetAdapter, error) {
+	server, err := grpc.NewGRPCServer(listeningPort)
+	if err != nil {
+		return nil, err
+	}
+	adapter := NetAdapter{
+		server: server,
+	}
+
+	newConnectionHandler := adapter.buildNewConnectionHandler()
+	server.SetNewConnectionHandler(newConnectionHandler)
+
+	return &adapter, nil
+}
+
+func (na *NetAdapter) buildNewConnectionHandler() func(connection server.Connection) {
```

Comment:
Actually, since it returns a hardcoded function, why can't this just be this function and that's it?
Why does the function need a constructor?

Reply:
> Why the build prefix?
I didn't want to call it newNewConnectionHandler, and since it's a constructor, I didn't want to omit the verb as you do with e.g. getters in Go.
> Why does the function need a constructor?
It needs `(na *NetAdapter)`.